File: benchmarks/1_sum/bench.c
/* Benchmark 1: Integer Sum — sum 0..999_999_999 (1B additions) */
#include <stdio.h>
#include <time.h>

int main(void) {
    clock_t t0 = clock();

    long long sum = 0;
    for (long long i = 0; i < 1000000000LL; i++) {
        sum += i;
    }

    clock_t t1 = clock();
    long long ms = (long long)((double)(t1 - t0) / CLOCKS_PER_SEC * 1000.0);

    printf("%lld\n", sum);
    printf("TIME_MS:%lld\n", ms);
    return 0;
}

File: tauraroc-Linux-X64/benchmarks/1_sum/bench.rs
// Benchmark 1: Integer Sum — sum 0..999_999_999 (1B additions)
use std::time::Instant;

fn main() {
    let t0 = Instant::now();

    let mut sum: i64 = 0;
    for i in 0i64..1_000_000_000 {
        sum += i;
    }

    let ms = t0.elapsed().as_millis();
    println!("{}", sum);
    println!("TIME_MS:{}", ms);
}
